ui/src/app/routing/workspace-app-routing.tsx

```diff
@@ -113,109 +113,112 @@ export const WorkspaceRoutes = React.memo(() => {
     </AppRoute>
-    <AppRoute path={`${path}/data/cohorts/:cid/review/participants`}>
+    <AppRoute exact path={`${path}/data/cohorts/:cid/review/participants`}>
       <ParticipantsTablePage routeData={{
         title: 'Review Cohort Participants',
         breadcrumb: BreadcrumbType.Cohort,
         workspaceNavBarTab: 'data',
         pageKey: 'reviewParticipants'
       }}/>
     </AppRoute>
-    <AppRoute path={`${path}/data/cohorts/:cid/review/participants/:pid`}>
+    <AppRoute exact path={`${path}/data/cohorts/:cid/review/participants/:pid`}>
       <DetailPagePage routeData={{
         title: 'Participant Detail',
         breadcrumb: BreadcrumbType.Participant,
         workspaceNavBarTab: 'data',
         pageKey: 'reviewParticipantDetail'
       }}/>
     </AppRoute>
-    <AppRoute path={`${path}/data/cohorts/:cid/review/cohort-description`}>
+    <AppRoute exact path={`${path}/data/cohorts/:cid/review/cohort-description`}>
       <QueryReportPage routeData={{
         title: 'Review Cohort Description',
         breadcrumb: BreadcrumbType.Cohort,
         workspaceNavBarTab: 'data',
         pageKey: 'cohortDescription'
       }}/>
     </AppRoute>
-    <AppRoute path={`${path}/data/cohorts/:cid/review`}>
+    <AppRoute exact path={`${path}/data/cohorts/:cid/review`}>
       <CohortReviewPage routeData={{
         title: 'Review Cohort Participants',
         breadcrumb: BreadcrumbType.Cohort,
         workspaceNavBarTab: 'data',
         pageKey: 'reviewParticipants'
       }}/>
     </AppRoute>
-    <AppRoute path={`${path}/data/concepts`}>
+    <AppRoute exact path={`${path}/data/concepts`}>
       <ConceptHomepagePage routeData={{
         title: 'Search Concepts',
         breadcrumb: BreadcrumbType.SearchConcepts,
         workspaceNavBarTab: 'data',
         pageKey: 'searchConceptSets'
       }}/>
     </AppRoute>
-    <AppRoute path={`${path}/data/concepts/sets/:csid`}>
+    <AppRoute exact path={`${path}/data/concepts/sets/:csid`}>
       <ConceptSearchPage routeData={{
         title: 'Concept Set',
         breadcrumb: BreadcrumbType.ConceptSet,
         workspaceNavBarTab: 'data',
         pageKey: 'conceptSets'
       }}/>
     </AppRoute>
-    <AppRoute path={`${path}/data/concepts/:domain`}>
+    <AppRoute exact path={`${path}/data/concepts/:domain`}>
       <ConceptSearchPage routeData={{
         title: 'Search Concepts',
         breadcrumb: BreadcrumbType.SearchConcepts,
         workspaceNavBarTab: 'data',
         pageKey: 'conceptSets'
       }}/>
     </AppRoute>
-    <AppRoute path={`${path}/data/concepts/sets/:csid/actions`}>
+    <AppRoute exact path={`${path}/data/concepts/sets/:csid/actions`}>
       <ConceptSetActionsPage routeData={{
         title: 'Concept Set Actions',
         breadcrumb: BreadcrumbType.ConceptSet,
         workspaceNavBarTab: 'data',
         pageKey: 'conceptSetActions'
       }}/>
     </AppRoute>
-  </React.Fragment>;
+    <AppRoute exact={false} path={`${path}`}>
```

Comment:
is this something we'll have to add to every nested route?

Reply:
Yes, otherwise it'll match the 'wrapper' component (and then it won't match of the wrapped routes and will spin forever)